inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/page/settings/WorkflowAndMonitoringPanel.java

```diff
@@ -85,28 +94,40 @@ public WorkflowAndMonitoringPanel(String aID, IModel<Project> aProject)
 
         //Finally, add the confirm button at the end
 
-        Button confirm = new AjaxButton(getString("confirm"),Model.of("Confirm")) {
-            @Override
-            public void onSubmit(AjaxRequestTarget target)
-            {
-                if (monitoringChoices.getDefaultModelObjectAsString().equals(monitoring.get(0)))
-                {
-                    workloadProperties.setActive(false);
-                } else {
-                    workloadProperties.setActive(true);
-                }
-
-                if (workflowChoices.getDefaultModelObjectAsString().equals(workflow.get(0)))
-                {
-                    workflowProperties.setActive(false);
-                } else {
-                    workflowProperties.setActive(true);
-                }
-
-                setResponsePage(getApplication().getHomePage());
-            }
-        };
+        Button confirm = new LambdaAjaxButton(getString("confirm"),
+            this::actionConfirm).triggerAfterSubmit();
 
         form.add(confirm);
+
+        add(form);
+    }
+
+    @Override
+    protected void onInitialize()
+    {
+        super.onInitialize();
+        IModel<String> resourceModel = new StringResourceModel(
+            getString(getId()), this, Model.of(getId()));
+
+    }
+
+    private void actionConfirm(AjaxRequestTarget aTarget, Form<?> aForm)
+    {
+        aTarget.add(getParent());
```

Comment:
Why do you need to add the parent?

Reply:
Yes just checked, it is not needed. I made it as it was done in webanno (e.g. LayerDetailForm). Removed it, thanks.